### DISCUSSION TITLE
Ktor Server Playground Default Route

### DISCUSSION THREAD

**Discussion by sangeetds** [ExpediaGroup/graphql-kotlin#1183] [Q&A] [answered]
Hello!

I was using the `graphql-kotlin-spring-server` with Ktor. I saw in the <a href="https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples/server/ktor-server">ktor example</a> that the default route for the playground was being handled manually, and a custom HTML page was being returned. But I observed that the spring server had few default routes already pre-configured. I wanted to ask whether there are simpler ways to generate the default routes in Ktor. Do I need to manually handle default endpoints? If yes, I would like to know if there's some documentation from where I can get more info about setting them up.

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
The reason for the differences between Spring and Ktor server examples is that we publish a Spring specific auto-configuration library that handles some of the common logic whereas Ktor example relies on a generic server logic. So unfortunately yes, you have to manually provide the logic for any of the endpoints in your Ktor app (as seen in the example app).

**Reply by smyrick** [reply]
If you would like to publish your own graphql-kotlin-ktor-server, which includes theses files and other features, we would be happy to link to it in our docs

**Reply by sangeetds** [reply]
Hi @smyrick. Sounds like an exciting prospect. I will surely try if I get time. Thanks a lot!
